lib/ProductOpener/APITest.pm

```diff
@@ -768,4 +771,60 @@ sub fake_http_server ($port, $dump_path, $responses_ref) {
 	return $httpd;
 }
 
+=head2 get_minion_jobs($task_name, $created_after_ts, $max_waiting_time)
+Subprogram which wait till the minion finished its job or
+if it takes too much time
+
+=head3 Arguments
+
+=head4 $task_name
+The name of the task 
+
+=head4 $created_after_ts
+The timestamp of the creation of the task
+
+=head4 $max_waiting_time
+The max waiting time for this given task
+
+=head3 Returns
+Returns a list of jobs associated with the task_name
+
+=cut
+
+sub get_minion_jobs ($task_name, $created_after_ts, $max_waiting_time) {
+	my $waited = 0;    # counting the waiting time
+	my %run_jobs = ();
+	while (($waited < $max_waiting_time) and (!(scalar %run_jobs))) {
+		my $jobs = get_minion()->jobs({tasks => [$task_name]});
+		# iterate on job
+		while (my $job = $jobs->next) {
+			next if (defined $run_jobs{$job->{id}});
+			# only those who were created after the timestamp
+			if ($job->{created} > $created_after_ts) {
+				# retrieving the job id
+				my $job_id = $job->{id};
+				# retrieving the job state
+				my $job_state = $job->{state};
+				# waiting the job to be done
+				while (($job_state eq "active") or ($job_state eq "inactive")) {
+					sleep(2);
+					$waited += 2;
+					# reload to get updated state
+					$job = get_minion()->job($job_id);
+					$job_state = $job->info->{state};
```

Comment:
Ok we may have a problem because $job when returned by `jobs` method returns a hash, while 'job' method return an object.

It's dangerous because, here it work for your test, but it might break in the future if you have multiple jobs for exemple (more chance the job is not waited for).

So I propose that we systematically get the object through job method. cf proposal.